src/app/modules/verification/verification.service.ts

```diff
@@ -611,20 +614,33 @@ const checkSmsCountAndPerformAction = (
 }
 
 /**
- * Check whether the field in the form is verifiable.
- * If it is not, then we don't need to generate an OTP.
+ * Check whether we should generate an OTP according to the requirements:
+ * 1. the field in the form is verifiable, and
+ * 2. if the recipient is within the allowed countries set by the field
+ * If these conditions are not met, then don't generate an OTP.
  */
 export const shouldGenerateMobileOtp = (
   { form_fields }: Pick<IFormSchema, 'form_fields'>,
   fieldId: string,
+  recipient: string,
 ): ResultAsync<true, OtpRequestError> => {
-  const isVerifiableMobileField =
-    !!form_fields &&
-    form_fields.filter(
-      ({ _id, isVerifiable }) => fieldId === String(_id) && isVerifiable,
-    ).length > 0
+  // Get the field with this fieldId
+  const field = form_fields?.find(({ _id }) => fieldId === String(_id))
+
+  if (!field)
+    return errAsync(new MalformedParametersError('Field id not found'))
+
+  if (field.fieldType !== BasicField.Mobile) {
+    return errAsync(
+      new MalformedParametersError('Field id is not a mobile field'),
```

Comment:
does returning `MalformedParametersError` change any error mapping in the controller?? do we return the expected error message when that happens?

Reply:
Minor nit that you should never see this case on the frontend (it's not possible for the field type to change). So maybe `OtpRequestError` is more appropriate

`if (field.fieldType !== BasicField.Mobile)`

Reply:
Or update the mapping to change the error message

Reply:
Sure, will make that change!